draft-ietf-avtcore-cc-feedback-message-01.txt

```diff
@@ -0,0 +1,616 @@
+   which it is reporting.  Following this, each sequence number between
+   the begin_seq and end_seq (both inclusive; modulo 65535 to account
+   for possible sequence number wrap-around) is represented by a 16-bit
+   packet metric block that contains the L, ECN, and ATO fields.  If the
+   number of 16-bit packet metric blocks included in the report block is
+   not a multiple of two, then 16 bits of zero padding MUST be added
+   after the last packet metric block, to align the end of the packet
+   metric blocks with the next 32 bit boundary.  In each packet metric
+   block, the L, ECN, and ATO fields are as follows:
+
+   o  L (1 bit): is a boolean to indicate if the packet was received. 0
+      represents that the packet was not yet received and all the
+      subsequent bits (ECN and ATO) are also set to 0.  1 represent the
+      packet was received and the subsequent bits in the block need to
+      be parsed.
+
+   o  ECN (2 bits): is the echoed ECN mark of the packet.  These are set
+      to 00 if not received, or if ECN is not used.
+
+   o  Arrival time offset (ATO, 13 bits): is the arrival time of the RTP
+      packet at the receiver.  It is measured as an offset from the time
+      at which the RTCP congestion control feedback report packet is
+      sent.  The arrival time offset is calculated by subtracting the
+      reception time of the RTP packet denoted by this 16 bit packet
+      metric block from the Report Timestamp (RTS) field of the RTCP
+      congestion control feedback report packet in which the packet
+      metric report block is contained.  The arrival time offset is
+      measured in units of 1/1024 seconds (this unit is chosen to give
+      exact offsets from the RTS field).  If the measured value is
+      greater than 8189/1024 seconds (the value that would be coded as
+      0x1FFD), the value 0x1FFE MUST be reported to indicate an over-
+      range positive measurement.  If the measurement is unavailable,
+      the value 0x1FFF MUST be reported.
+
+
+
+
+
+Sarker, et al.          Expires September 3, 2018               [Page 5]
+
+Internet-Draft     Congestion Control Feedback in RTCP        March 2018
+
+
+   The RTCP congestion control feedback report packet concludes with the
+   Report Timestamp field (RTS, 32 bits).  This represents the time
+   instant when the report packet was generated.  The value of RTS field
+   is derived from the same wallclock used to generate the NTP timestamp
+   field in RTCP Sender Report (SR) and Receiver Report (RR) packets.
+   It is formatted as the middle 32 bits of an NTP format timestamp, as
+   described in Section 4 of [RFC3550].
+
+   RTCP congestion control feedback packets SHOULD include a report
+   block for each SSRC that is being congestion controlled.  The
+   sequence number ranges reported on in consecutive reports for an SSRC
+   SHOULD be consecutive and SHOULD NOT overlap (i.e., begin_seq for a
```

Comment:
Since RTCP packets can be lost, wouldnt over-lapping sequence numbers help provide more information, even though they might be too late for congestion control to react upon. 

Although it depends on the timeout value for halving the sending rate...

Reply:
My feeling was that they'd be too late, but we should discuss at the meeting.

Reply:
I share Colin's feeling here